src/coreComponents/mesh/mpiCommunications/NeighborCommunicator.hpp

```diff
@@ -46,6 +46,32 @@ class NeighborCommunicator
   NeighborCommunicator():
     NeighborCommunicator( -1 ){};
 
+
+  enum class Location
+  {
+    Elem, //!< location is element (like pressure in finite volumes)
+    Face, //!< location is face (like flux in mixed finite elements)
+    Edge, //!< location is edge (like flux between fracture elements)
+    Node  //!< location is node (like displacements in finite elements)
+  };
+
+  struct SyncFieldsID
+  {
+    string regionName;
```

Comment:
Can this become a list of regions? Typical use case for any cell-centered (FV) solver will be to pass its whole list of `targetRegions` for a given mesh directly. We shouldn't force the solver to additionally iterate over regions and add repeated records with the same `fieldNames`.

Reply:
yes, sure. Some solvers have different dofs for each region type so they won't exactly pass the full list but I guess having a region list is indeed better. 

I am also open to suggestions about where to define this class and this `enum` so that they can also be used in the `DofManager` which currently calls that `syncrhonizeFields` as well and will need to provide a list of regions. However, the `DofManager` already has a struct that basically holds the same information of this one so it would make sense to use the same one instead of having to copy the information once again.